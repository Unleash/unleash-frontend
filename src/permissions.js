export const ADMIN = 'ADMIN';
export const CREATE_FEATURE = 'CREATE_FEATURE';
export const UPDATE_FEATURE = 'UPDATE_FEATURE';
export const DELETE_FEATURE = 'DELETE_FEATURE';
export const CREATE_STRATEGY = 'CREATE_STRATEGY';
export const UPDATE_STRATEGY = 'UPDATE_STRATEGY';
export const DELETE_STRATEGY = 'DELETE_STRATEGY';
export const UPDATE_APPLICATION = 'UPDATE_APPLICATION';
export const CREATE_CONTEXT_FIELD = 'CREATE_CONTEXT_FIELD';
export const UPDATE_CONTEXT_FIELD = 'UPDATE_CONTEXT_FIELD';
export const DELETE_CONTEXT_FIELD = 'DELETE_CONTEXT_FIELD';

export function hasPermission(user, permission) {
    return (
        user &&
        (!user.permissions || user.permissions.indexOf(ADMIN) !== -1 || user.permissions.indexOf(permission) !== -1)
    );
}
