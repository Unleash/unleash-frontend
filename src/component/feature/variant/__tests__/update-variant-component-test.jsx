import React from 'react';
import { MemoryRouter } from 'react-router-dom';

import UpdateVariant from './../update-variant-component';
import renderer from 'react-test-renderer';
import { UPDATE_FEATURE } from '../../../../permissions';
import { weightTypes } from '../enums';

jest.mock('react-mdl');

test('renders correctly with without variants', () => {
    const tree = renderer.create(
        <MemoryRouter>
            <UpdateVariant
                key={0}
                variants={[]}
                addVariant={jest.fn()}
                removeVariant={jest.fn()}
                updateVariant={jest.fn()}
                hasPermission={permission => permission === UPDATE_FEATURE}
            />
        </MemoryRouter>
    );

    expect(tree).toMatchSnapshot();
});

test('renders correctly with without variants and no permissions', () => {
    const tree = renderer.create(
        <MemoryRouter>
            <UpdateVariant
                key={0}
                variants={[]}
                addVariant={jest.fn()}
                removeVariant={jest.fn()}
                updateVariant={jest.fn()}
                hasPermission={() => false}
            />
        </MemoryRouter>
    );

    expect(tree).toMatchSnapshot();
});

test('renders correctly with with variants', () => {
    const featureToggle = {
        name: 'toggle.variants',
        description: 'description',
        enabled: false,
        strategies: [
            {
                name: 'gradualRolloutRandom',
                parameters: {
                    percentage: 50,
                },
            },
        ],
        variants: [
            {
                name: 'blue',
                weight: 34,
                overrides: [
                    {
                        field: 'userId',
                        values: ['1337', '123'],
                    },
                ],
            },
            {
                name: 'yellow',
                weight: 33,
            },
            {
                name: 'orange',
                weight: 33,
                weightType: weightTypes.FIX,
                payload: {
                    type: 'string',
                    value: '{"color": "blue", "animated": false}',
                },
            },
        ],
        createdAt: '2018-02-04T20:27:52.127Z',
    };
    const tree = renderer.create(
        <MemoryRouter>
            <UpdateVariant
                key={0}
                variants={featureToggle.variants}
                addVariant={jest.fn()}
                removeVariant={jest.fn()}
                updateVariant={jest.fn()}
                hasPermission={permission => permission === UPDATE_FEATURE}
            />
        </MemoryRouter>
    );

    expect(tree).toMatchSnapshot();
});
