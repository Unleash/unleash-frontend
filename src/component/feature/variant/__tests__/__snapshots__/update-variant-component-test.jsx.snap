// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with with variants 1`] = `
<section
  style={
    Object {
      "maxWidth": "700px",
      "padding": "16px",
    }
  }
>
  <p>
    Variants allows you to return a variant object if the feature toggle is considered enabled for the current request. When using variants you should use the
     
    <code
      style={
        Object {
          "color": "navy",
        }
      }
    >
      getVariant()
    </code>
     method in the Client SDK.
  </p>
  <table
    className="mdl-data-table mdl-shadow--2dp variantTable"
  >
    <thead>
      <tr>
        <th>
          Variant name
        </th>
        <th
          className="labels"
        />
        <th>
          Weight
        </th>
        <th>
          Weight Type
        </th>
        <th
          className="actions"
        />
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          onClick={[Function]}
        >
          blue
        </td>
        <td
          className="labels"
        >
           
          <react-mdl-Chip
            style={
              Object {
                "backgroundColor": "rgba(173, 216, 230, 0.2)",
              }
            }
          >
            Overrides
          </react-mdl-Chip>
        </td>
        <td>
          3.4
           %
        </td>
        <td>
          Variable
        </td>
        <td
          className="actions"
        >
          <react-mdl-IconButton
            name="edit"
            onClick={[Function]}
          />
          <react-mdl-IconButton
            name="delete"
            onClick={[Function]}
          />
        </td>
      </tr>
      <tr>
        <td
          onClick={[Function]}
        >
          yellow
        </td>
        <td
          className="labels"
        >
           
        </td>
        <td>
          3.3
           %
        </td>
        <td>
          Variable
        </td>
        <td
          className="actions"
        >
          <react-mdl-IconButton
            name="edit"
            onClick={[Function]}
          />
          <react-mdl-IconButton
            name="delete"
            onClick={[Function]}
          />
        </td>
      </tr>
      <tr>
        <td
          onClick={[Function]}
        >
          orange
        </td>
        <td
          className="labels"
        >
          <react-mdl-Chip>
            Payload
          </react-mdl-Chip>
           
        </td>
        <td>
          3.3
           %
        </td>
        <td>
          Fix
        </td>
        <td
          className="actions"
        >
          <react-mdl-IconButton
            name="edit"
            onClick={[Function]}
          />
          <react-mdl-IconButton
            name="delete"
            onClick={[Function]}
          />
        </td>
      </tr>
    </tbody>
  </table>
  <br />
  <p>
    <a
      href="#add-variant"
      onClick={[Function]}
      title="Add variant"
    >
      Add variant
    </a>
  </p>
</section>
`;

exports[`renders correctly with without variants 1`] = `
<section
  style={
    Object {
      "maxWidth": "700px",
      "padding": "16px",
    }
  }
>
  <p>
    Variants allows you to return a variant object if the feature toggle is considered enabled for the current request. When using variants you should use the
     
    <code
      style={
        Object {
          "color": "navy",
        }
      }
    >
      getVariant()
    </code>
     method in the Client SDK.
  </p>
  <p>
    No variants defined.
  </p>
  <br />
  <p>
    <a
      href="#add-variant"
      onClick={[Function]}
      title="Add variant"
    >
      Add variant
    </a>
  </p>
</section>
`;

exports[`renders correctly with without variants and no permissions 1`] = `
<section
  style={
    Object {
      "maxWidth": "700px",
      "padding": "16px",
    }
  }
>
  <p>
    Variants allows you to return a variant object if the feature toggle is considered enabled for the current request. When using variants you should use the
     
    <code
      style={
        Object {
          "color": "navy",
        }
      }
    >
      getVariant()
    </code>
     method in the Client SDK.
  </p>
  <p>
    No variants defined.
  </p>
  <br />
</section>
`;
